cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/TestModule2.java

```diff
@@ -41,11 +42,17 @@ public void register(DatasetDefinitionRegistry registry) {
     registry.add(createDefinition("datasetType2"));
   }
 
-  private DatasetDefinition createDefinition(String name) {
+  public DatasetDefinition createDefinition(String name) {
     return new AbstractDatasetDefinition(name) {
       @Override
       public DatasetSpecification configure(String instanceName, DatasetProperties properties) {
-        return createSpec(instanceName, getName(), properties);
+        return DatasetSpecification
+          .builder(instanceName, getName())
+          .properties(ImmutableMap.<String, String>builder()
+                        .putAll(properties.getProperties())
+                        .put("extra", "value")
```

Comment:
this is to test that the dataset summary returned by list() contains the original properties. Original properties are different from the spec.getProperties() if the configure() method modifies the original properties. That is what we mimic here.